.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -20,6 +21,16 @@ RUN spack compiler info $COMPILER > /dev/null 2> /dev/null; compiler_missing=$?;
     if [[ $compiler_missing != 0 ]]; then \
         spack install $spack_compiler arch=$ARCH && \
         spack compiler add --scope site $(spack location -i $spack_compiler) && \
+        if [[ ! -z "$NVHPC_COMPILER" ]]; then \
+            # Require the spack compilers instead of the system compiler
+            printf \
+            "  all:\n    require:\n    - any_of: [\"\045${NVHPC_COMPILER}\",\"\045${COMPILER}\"]" \
```

Comment:
Just wondering: what is the `\045` for?

Reply:
I think you can do the same as in DLA-Future: https://github.com/eth-cscs/DLA-Future/blob/70dd621e8038bba67d69be9f7e6499c4b44f7eea/ci/docker/build.Dockerfile#L55.

Reply:
About the `spack config add`, I think it is a limit of the CLI tool. I found also other comments about it, like https://spackpm.slack.com/archives/C5W7NKZJT/p1672863096551769.

The problem I see is that `require:` has to be an array, and `any_of` has to be an array again. So it should look like `packages:all:require:[any_of:["%gcc", "%nvcc]]`, and the parser is not so advanced to parse similar syntax.